### PR TITLE
UIDATIMP-1662: Adjust value for invoice and invoice lines adjustments "Type" field mapping

### PR DESCRIPTION
**Pull request by OleksandrHladchenko1**
## Purpose

 * Adjust the value for invoice and invoice lines adjustments type field mapping (mapping entries with property "path" = "invoice.adjustments[].type" and "path" = "invoice.invoiceLines[].adjustments[].type"):

      - capitalize the first letter of the value 
      - wrap the value with escaped quotes 
* When this PR is merged we'll receive the data form BE in format `'\"Amount\"'` instead of `'amount'`.

## Approach

* Update version of actions/upload-artifact to v4 due to failed build

* Change the form value when user interacts with `TypeToggle` component
* Updated default schema for invoice field mapping profile.
* Update the behavior of `renderAmountValue` method.
* Added unit tests.

## Refs

[]()

## Screenshots

https://github.com/user-attachments/assets/178ca1a9-ccb3-4157-bc36-ac9786e58922

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@
 * Transform `ViewAllLogs` component into the functional one (UIDATIMP-1630)
 * Change endpoint to get actual SRS MARC data (UIDATIMP-1597)
 * Implement new Donor information section for Order field mapping profiles (UIDATIMP-1656)
+* Adjust value for invoice and invoice lines adjustments "Type" field mapping. (uidatimp-1662)
```
Comment (mariia-aloshyna) [inline CHANGELOG.md:13]:
```suggestion
* Adjust value for invoice and invoice lines adjustments "Type" field mapping. (UIDATIMP-1662)
```

File: src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/InvoiceAdjustments.js [hunk 2]
```diff
@@ -272,6 +282,8 @@ export const InvoiceAdjustments = ({
               label={<FormattedMessage id={`${TRANSLATION_ID_PREFIX}.invoice.invoiceAdjustments.field.type`} />}
               name={INVOICE_ADJUSTMENTS_FIELDS_MAP.TYPE(index)}
               currency={currency}
+              onChangeToPercent={onChangeToPercent}
+              onChangeToAmount={onChangeToAmount}
```
Comment (mariia-aloshyna) [inline src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/InvoiceAdjustments.js:286]: why `format()` `parse()` can't be used here? In such case, we wouldn't need changes in stripes-acq-components

File: src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/tests/InvoiceAdjustments.test.js [hunk 3]
```diff
@@ -25,6 +25,7 @@ jest.mock('../../../hooks', () => ({
 
 global.fetch = jest.fn();
 
+const invoiceAdjustmentsTypePath = 'profile.mappingDetails.mappingFields[15].subfields.0.fields.2.value';
```
Comment (mariia-aloshyna) [inline src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/tests/InvoiceAdjustments.test.js:28]: use `getSubfieldName()` to get path name, please

File: src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/tests/InvoiceLineAdjustments.test.js [hunk 4]
```diff
@@ -25,13 +22,17 @@ jest.mock('../../../hooks', () => ({
   useFieldMappingRefValues: () => [[mockLineAdjustmentsFields]],
 }));
 
+const setReferenceTablesMock = jest.fn();
 const okapi = buildOkapi();
 
+const invoiceLineAdjustmentsPath = 'profile.mappingDetails.mappingFields[27].subfields.0.fields.15.value';
```
Comment (mariia-aloshyna) [inline src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/tests/InvoiceLineAdjustments.test.js:28]: here as well

File: src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/tests/InvoiceLineAdjustments.test.js [hunk 5]
```diff
@@ -25,13 +22,17 @@ jest.mock('../../../hooks', () => ({
...
+const invoiceLineAdjustmentsTypePath = 'profile.mappingDetails.mappingFields[27].subfields.0.fields.15.subfields.0.fields.2.value';
```
Comment (mariia-aloshyna) [inline src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/tests/InvoiceLineAdjustments.test.js:29]: here `getInnerSubfieldName()` can be used

File: src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/tests/InvoiceLineAdjustments.test.js [hunk 6]
```diff
@@ -76,5 +81,38 @@ describe('InvoiceLineAdjustments edit component', () => {
       expect(getByText('Relation to total')).toBeInTheDocument();
       expect(getByText('Export to accounting')).toBeInTheDocument();
     });
+
+    describe('when click on trash icon', () => {
+      it('shold call the function to clean fields', () => {
+        const { container } = renderInvoiceLineAdjustments();
+
+        const deleteButton = container.querySelector('[data-test-repeatable-field-remove-item-button="true"]');
```
Comment (mariia-aloshyna) [inline src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/tests/InvoiceLineAdjustments.test.js:89]: can `getByRole()` be used to find the trash icon?

File: src/settings/MappingProfiles/detailsSections/utils.js [hunk 7]
```diff
@@ -242,7 +242,7 @@ export const getAccountNumbersByCode = organization => {
 };
 
 export const renderAmountValue = (amountValue, amountType, currency) => {
-  const amountSymbol = amountType === '"percentage"' ?
+  const amountSymbol = amountType === '"Percentage"' ?
```
Comment (mariia-aloshyna) [inline src/settings/MappingProfiles/detailsSections/utils.js:245]: let's move `Percentage` and `Amoubt` this to constants
```suggestion
  const amountSymbol = amountType === `"${PERCENTAGE_VALUE}"` ?
```

File: src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/InvoiceAdjustments.js [hunk 8]
```diff
@@ -57,6 +58,7 @@ import {
   CURRENCY_FIELD,
 } from '../../../../../utils';
 
+
```
Comment (mariia-aloshyna) [inline src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/InvoiceAdjustments.js:61]: remove the extra line, please

**Review by mariia-aloshyna — APPROVED** [review]

File: src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/InvoiceLineAdjustments.js [hunk 9]
```diff
@@ -18,6 +18,7 @@ import {
 } from '@folio/stripes/components';
 import { TypeToggle } from '@folio/stripes-acq-components';
 
+import { capitalize } from 'lodash';
```
Comment (BogdanDenis) [inline src/settings/MappingProfiles/detailsSections/edit/InvoiceDetailSection/InvoiceLineAdjustments.js:21]: Please move this import higher

**Review by BogdanDenis — APPROVED** [review]

**Pull request merged**